library/connect.c

```diff
@@ -572,21 +576,19 @@ static void ziti_write_timeout(uv_timer_t *t) {
     uv_close((uv_handle_t *) t, free_handle);
 }
 
-static void ziti_write_async(uv_async_t *ar) {
-    struct ziti_write_req_s *req = ar->data;
+static void ziti_write_req(struct ziti_write_req_s *req) {
     struct ziti_conn *conn = req->conn;
 
-    if (conn->state >= Disconnected) {
+    if (conn->state >= Timedout) {
```

Comment:
why not go down to `CloseWrite`?

Reply:
`CloseWrite` can be set while there is still data in the queue. FIN will be sent after all data is sent